usaspending_api/api_contracts/contracts/v2/download/accounts.md

```diff
@@ -94,9 +94,24 @@ Generate files and return metadata using filters on custom account
         + `award_financial`
 + `fy` (required, string)
     The fiscal year to filter by in the format `YYYY`
-+ `quarter` (required, enum[string])
++ `quarter` (optional, enum[string])
+    Either `quarter` or `period` is required.  Do not supply both.   Note that both monthly and quarterly submissions will be included in the resulting download file even if only `quarter` is provided.
     + Members
         + `1`
         + `2`
         + `3`
         + `4`
++ `period` (optional, enum[string])
+    Either `quarter` or `period` is required.  Do not supply both.  Agencies cannot submit data for period 1 so it is disallowed as a query filter.   Note that both monthly and quarterly submissions will be included in the resulting download file even if only `period` is provided.
+    + Members
```

Comment:
@kbard 

Is there no period 1?

Also, can you confirm the return implicitly includes previous periods w/in the same FY?

For example, if I request period 4, it will return data for all periods in that FY less than or equal to period 4.

Reply:
There is no period 1.

As for your second question, the logic is fairly complicated.  From in-code comments:
```
# For Files A and B, we only care about the specific period or quarter requested; e.g. "get
# me Period 4 monthly submissions AND Quarter 1 quarterly submissions".  This is because
# monthly and quarterly submissions are on different tracks but both need to be included in
# the download file.

# For File C, we include rows in the specific period/quarter requested AND rows from
# previous periods or quarters where the transaction_obligated_amount is non-zero.
```

So the answer is "it depends".  Some downloads return previous periods some do not.

Reply:
So to follow-up on this from a Slack call Max and I had, since our current use case is always "from the beginning of the FY up to and including the quarter/period selected", all the front end needs to return is the FY and quarter or period for all endpoints that use the period/period picker.  The endpoint will handle the logic for deciding whatever needs to be returned.